consensus/consortium/v2/consortium_test.go

```diff
@@ -2274,18 +2294,251 @@ func TestIsPeriodBlock(t *testing.T) {
 	// this must be a period block
 	header = bs[399].Header()
 	// this header must be period header
-	if !c.IsPeriodBlock(chain, header, nil) {
+	isPeriodBlock, err = c.IsPeriodBlock(chain, header, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !isPeriodBlock {
 		t.Errorf("wrong period block")
 	}
 
 	// header of block 500
 	// this must not be a period block
 	header = bs[499].Header()
-	if c.IsPeriodBlock(chain, header, nil) {
+	isPeriodBlock, err = c.IsPeriodBlock(chain, header, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if isPeriodBlock {
 		t.Errorf("wrong period block")
 	}
 }
 
+func generateChain(
```

Comment:
Could we add some description/purpose for this method with flag `bumpTimeStampLastBlockOfEpoch` usage for setting the timestamp of  lastBlock